train.py

```
@@ -369,100 +452,121 @@
                         scheduler.step(val_loss_eval)
+
                     current_eval_loss = val_loss_eval
                     if val_loss_eval < best_val_loss:
                         best_val_loss = val_loss_eval
                         is_best_step = True
-                save_checkpoint(model, optimizer, epoch, global_step, current_eval_loss, args, is_best_step, scheduler)
 
-        # End of epoch
+                # Synchronize before saving
+                dist.barrier()
+                save_checkpoint(model, optimizer, epoch, global_step, current_eval_loss, args, is_best_step, scheduler, rank)
```

Pylint (3.3.7)

[convention] Line too long (125/100)

(C0301)